### PR TITLE
feat: "create personal account" - add "username" and "summary" UI [AR-1069]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1069" title="AR-1069" target="_blank">SubtaskAR-1069</a>  implement the registration personal flow UI
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

User needs to be able to fill out a personal account creation flow.
Steps needs to be reusable for both cases: create personal account and create team.

### Solutions

- implemented "enter username" screen UI as a new `NavigationItem`
- implemented "summary" screen UI as a new `NavigationItem`
- added `CreateAccountFlowType` to check which flow user is in and show appropriate title, label and image
- fixed incorrect steps screens packages
- fixed navigating back after navigating with `CLEAR_TILL_START` [[AR-1208]](https://wearezeta.atlassian.net/browse/AR-1208)
- fixed generic "server error" instead of "invalid password" when logging in and deleting client [[AR-1215]](https://wearezeta.atlassian.net/browse/AR-1215) [[AR-1207]](https://wearezeta.atlassian.net/browse/AR-1207)

### Attachments (Optional)

https://user-images.githubusercontent.com/30429749/157281052-ab74aa0b-8b44-4ae1-a67e-a3f4c11b0a0e.mp4

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[AR-1208]: https://wearezeta.atlassian.net/browse/AR-1208?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/CreateAccountFlowType.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        summaryTextResId = R.string.create_personal_account_summary_text,
+        summaryIconResId = R.drawable.ic_create_personal_account_success,
+    ),
+    CreateTeam(
+        routeArg = "create_team",
+        titleResId = R.string.create_personal_account_title, //TODO change
+        summaryTextResId = R.string.create_personal_account_summary_text, //TODO change
+        summaryIconResId = R.drawable.ic_create_personal_account_success, //TODO change
+        ),
+    None(titleResId = R.string.set_username_title);
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/authentication/create/CreateAccountFlowType.kt:25]: I don't understand why None is used here for

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/CreateAccountFlowType.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import com.wire.android.R
+
+enum class CreateAccountFlowType(
+    val routeArg: String = "",
+    @StringRes val titleResId: Int,
+    @StringRes val summaryTextResId: Int = 0,
+    @DrawableRes val summaryIconResId: Int = 0,
```
Comment (MohamadJaara) [inline app/src/main/kotlin/com/wire/android/ui/authentication/create/CreateAccountFlowType.kt:11]: Having initial values here is unitary + calling stringResource({ResIdThatDoesNotExist}) will throw an exception (I think 😅)
```suggestion
    val routeArg: String,
    @StringRes val titleResId: Int,
    @StringRes val summaryTextResId: Int,
    @DrawableRes val summaryIconResId: Int
```

**Review by MohamadJaara — CHANGES_REQUESTED** [review]
A question about `CreateAccountFlowType.None` and suggesting not to allow navigating back after a user already registered/logged in

File: app/src/main/kotlin/com/wire/android/ui/authentication/devices/RemoveDeviceViewModel.kt [hunk 3]
```diff
@@ -112,7 +115,7 @@ class RemoveDeviceViewModel @Inject constructor(
             is RegisterClientResult.Failure.InvalidCredentials -> RemoveDeviceError.InvalidCredentialsError
             is RegisterClientResult.Failure.TooManyClients -> RemoveDeviceError.TooManyDevicesError
             is RegisterClientResult.Success -> RemoveDeviceError.None
-            else -> RemoveDeviceError.None
+            else -> RemoveDeviceError.GenericError(CoreFailure.Unknown(IllegalStateException()))
```
[the remark this answers is not in the document]
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/authentication/devices/RemoveDeviceViewModel.kt]: Someone added `else` so I just changed it not to be successful (error None means success).
For some reason Android Studio shows an error without `else` even when all cases are included:
image
There is a bug reported already: https://youtrack.jetbrains.com/issue/KTIJ-1270
But it doesn't affect the compiler so I'll remove `else` 🚀

File: app/src/main/kotlin/com/wire/android/ui/authentication/devices/RemoveDeviceViewModel.kt [hunk 4]
```diff
@@ -102,8 +104,9 @@ class RemoveDeviceViewModel @Inject constructor(
     private fun DeleteClientResult.toRemoveDeviceError() =
         when (this) {
             is DeleteClientResult.Failure.Generic -> RemoveDeviceError.GenericError(this.genericFailure)
-            is DeleteClientResult.Success -> RemoveDeviceError.None
-            else -> RemoveDeviceError.None
+            DeleteClientResult.Failure.InvalidCredentials -> RemoveDeviceError.InvalidCredentialsError
+            DeleteClientResult.Success -> RemoveDeviceError.None
+            else -> RemoveDeviceError.GenericError(CoreFailure.Unknown(IllegalStateException()))
```
[the remark this answers is not in the document]
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/authentication/devices/RemoveDeviceViewModel.kt]: Same as above 🚀

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/username/CreateAccountUsernameViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+class CreateAccountUsernameViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+    private val navigationManager: NavigationManager
+): ViewModel() {
+    private val type: CreateAccountFlowType =
+        CreateAccountFlowType.fromString(savedStateHandle.getLiveData<String>(EXTRA_CREATE_ACCOUNT_FLOW_TYPE).value)
+    var state: CreateAccountUsernameViewState by mutableStateOf(CreateAccountUsernameViewState(type))
+        private set
+
+    fun navigateBack() { viewModelScope.launch { navigationManager.navigateBack() } }
```
[the remark this answers is not in the document]
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/authentication/create/username/CreateAccountUsernameViewModel.kt]: I removed it and blocked "back" click  🚀

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/CreateAccountFlowType.kt (hunk 2 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/authentication/create/CreateAccountFlowType.kt]: Right, fixed it 🚀

File: app/src/main/kotlin/com/wire/android/ui/authentication/create/CreateAccountFlowType.kt (hunk 1 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/authentication/create/CreateAccountFlowType.kt]: I separated both situations and changed it to `AppStart` - the situation where app detects at the start that this user haven't set his handle yet 🚀

**Review by MohamadJaara — APPROVED** [review]
Nicely done 🚀

**Pull request merged**